src/module/item/spell/document.ts

```diff
@@ -150,11 +150,12 @@ class SpellPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Ite
     }
 
     get isCantrip(): boolean {
-        return this.traits.has("cantrip") && !this.isRitual;
+        return this.system.traits.value.includes("cantrip");
     }
 
     get isFocusSpell(): boolean {
-        return this.traits.has("focus");
+        const traits = this._source.system.traits;
+        return traits.traditions.length === 0 && (this.isCantrip || traits.value.includes("focus"));
```

Comment:
```suggestion
        return (traits.traditions.length === 0 && this.isCantrip) || traits.value.includes("focus");
```
if you don't want to give tikael a heart attack in the meantime